File: frontend/src/services/filmService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/films';

export const getFilms = (category) => axios.get(API_URL, {
    params: { category }
});
export const getFilmByEidr = (eidr) => axios.get(`${API_URL}/${eidr}`);
export const addFilm = (film) => axios.post(API_URL, film);
export const updateFilmStatus = (eidr, active) => axios.put(`${API_URL}/${eidr}`, null, { params: { active } });
export const deleteFilm = (eidr) => axios.delete(`${API_URL}/${eidr}`);

File: backend/src/main/java/org/example/service/FilmService.java
package org.example.service;

import org.example.model.Film;

import java.util.List;

public interface FilmService {
    List<Film> getAllFilms(String category);
    Film getFilmByEidr(String eidr);
    void addFilm(Film film);
    void updateFilmStatus(String eidr, boolean active);
    void deleteFilm(String eidr);
}